src/publishers/zipDeploy.ts

```diff
@@ -149,8 +149,8 @@ export class ZipDeploy {
         while (retryCount > 0) {
             await Sleeper.timeout(retryInterval);
             try {
-                const settings: any = await context.kuduService.getAppSettings();
-                if (settings && settings[key] && settings[key] === expectedValue) {
+                const settings: any = await context.appService.getApplicationSettings();
```

Comment:
Could you make this into a back-filling logic? If the customer uses ScmCredentials for deployment, the appService will become undefined. So a proper way to do this is something like:
```javascript
let settings: any = await context.kuduService.getAppSettings();
if (!settings) {
   settings = await context.appService.getApplicationSettings();
}
```